SampleLibrary/iOS/UIAddViewController.swift

```diff
@@ -0,0 +1,117 @@
+		if userCancelled {
+			completionBlock(.failure(NSError.userCancelledError()))
+		}
+		
+	}
+	
+	/// This function sets the UI components' properties, such as constraints, colors, texts, etc.
+	private func setUpView(){
+		func setUpTextField(textField: UITextField, placeholder placeholderText: String, defautValue value: String){
+			textField.backgroundColor = .systemPurple.withAlphaComponent(0.2)
+			textField.layer.cornerRadius = 10
+			textField.layer.masksToBounds = true
+			textField.textAlignment = .center
+			textField.keyboardType = .numberPad
+			textField.leftViewMode = .always
+			let placeholder = UILabel()
+			placeholder.text = placeholderText
+			textField.leftView = placeholder
+			textField.text = value
+		}
+		
+		view.backgroundColor = .black
+		
+		setUpTextField(textField: firstAddends, placeholder:" Number 1:", defautValue: String(firstNumber))
+		view.addSubview(firstAddends)
+		firstAddends.translatesAutoresizingMaskIntoConstraints = false
+		firstAddends.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor, constant: padding).isActive = true
+		firstAddends.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: padding).isActive = true
+		firstAddends.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -padding).isActive = true
+		firstAddends.heightAnchor.constraint(equalToConstant: 50).isActive = true
+		
+		setUpTextField(textField: secondAddends, placeholder:" Number 2:", defautValue: String(secondNumber))
+		view.addSubview(secondAddends)
+		secondAddends.translatesAutoresizingMaskIntoConstraints = false
+		secondAddends.topAnchor.constraint(equalTo: firstAddends.bottomAnchor, constant: padding).isActive = true
+		secondAddends.leadingAnchor.constraint(equalTo: firstAddends.leadingAnchor).isActive = true
+		secondAddends.trailingAnchor.constraint(equalTo: firstAddends.trailingAnchor).isActive = true
+		secondAddends.heightAnchor.constraint(equalToConstant: 50).isActive = true
+		
+		view.addSubview(sumButton)
+		sumButton.translatesAutoresizingMaskIntoConstraints = false
+		sumButton.topAnchor.constraint(equalTo: secondAddends.bottomAnchor, constant: padding).isActive = true
+		sumButton.centerXAnchor.constraint(equalTo: view.centerXAnchor).isActive = true
+		sumButton.widthAnchor.constraint(equalToConstant: 100).isActive = true
+		sumButton.heightAnchor.constraint(equalToConstant: 40).isActive = true
+		
+		sumButton.setTitle("Sum", for: .normal)
+		sumButton.backgroundColor = .systemPurple
+		sumButton.setTitleColor(.white, for: .normal)
+		sumButton.layer.cornerRadius = 20
+		sumButton.addTarget(self, action: #selector(didPressSumButton(_:)), for: .touchUpInside)
+	}
+	
+	@objc func didPressSumButton(_ sender: UIButton) {
+		// Setting userCancelled = false because we are going to dismiss the UIViewController because the user pressed the sum button and did not close the UIViewController
+		userCancelled = false
```

Comment:
Instead of using a `userCancelled` flag, `completionBlock` var could be optional, and set to nil before being called, ensuring it will be called only once.

Reply:
It's a clever way to use the completionBlock value as a flag, I like it 👌